cmd/kubeadm/app/util/apiclient/wait.go

```diff
@@ -235,12 +233,9 @@ func getStaticPodSingleHash(client clientset.Interface, nodeName string, compone
 		return "", err
 	}
 
-	podBytes, err := json.Marshal(staticPod)
-	if err != nil {
-		return "", err
-	}
-
-	return fmt.Sprintf("%x", sha256.Sum256(podBytes)), nil
+	staticPodHash := staticPod.Annotations["kubernetes.io/config.hash"]
```

Comment:
@detiber One minor here: what about using the constant kubetypes.ConfigHashAnnotationKey instead of the annotation name? Doing in this way there will be a signal if something changes in kubelet...

Reply:
+1 there ^